src/config_server.c

```diff
@@ -0,0 +1,2396 @@
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "compat.h"
+#include "libnetconf.h"
+#include "session_server.h"
+#include "session_server_ch.h"
+
+/* All libssh supported host-key, key-exchange, encryption and mac algorithms as of version 0.10.90 */
+
+static const char *supported_hostkey_algs[] = {
+    "ssh-ed25519-cert-v01@openssh.com", "ecdsa-sha2-nistp521-cert-v01@openssh.com",
+    "ecdsa-sha2-nistp384-cert-v01@openssh.com", "ecdsa-sha2-nistp256-cert-v01@openssh.com",
+    "rsa-sha2-512-cert-v01@openssh.com", "rsa-sha2-256-cert-v01@openssh.com",
+    "ssh-rsa-cert-v01@openssh.com", "ssh-dss-cert-v01@openssh.com",
+    "ssh-ed25519", "ecdsa-sha2-nistp521", "ecdsa-sha2-nistp384", "ecdsa-sha2-nistp256",
+    "rsa-sha2-512", "rsa-sha2-256", "ssh-rsa", "ssh-dss", NULL
+};
+
+static const char *supported_kex_algs[] = {
+    "diffie-hellman-group-exchange-sha1", "curve25519-sha256", "curve25519-sha256@libssh.org",
+    "ecdh-sha2-nistp256", "ecdh-sha2-nistp384", "ecdh-sha2-nistp521", "diffie-hellman-group18-sha512",
+    "diffie-hellman-group16-sha512", "diffie-hellman-group-exchange-sha256", "diffie-hellman-group14-sha256", NULL
+};
+
+static const char *supported_encryption_algs[] = {
+    "chacha20-poly1305@openssh.com", "aes256-gcm@openssh.com", "aes128-gcm@openssh.com",
+    "aes256-ctr", "aes192-ctr", "aes128-ctr", "aes256-cbc", "aes192-cbc", "aes128-cbc",
+    "blowfish-cbc", "3des-cbc", "none", NULL
+};
+
+static const char *supported_mac_algs[] = {
+    "hmac-sha2-256-etm@openssh.com", "hmac-sha2-512-etm@openssh.com", "hmac-sha1-etm@openssh.com",
+    "hmac-sha2-256", "hmac-sha2-512", "hmac-sha1", NULL
+};
+
+extern struct nc_server_opts server_opts;
+
+/**
+ * @brief Get the pointer to an endpoint structure based on node's location in the YANG data.
+ *
+ * @param[in] node Node from which the endpoint containing this node is derived.
+ * @param[out] endpt Endpoint containing the node.
+ * @param[out] bind Bind corresponding to the endpoint. Optional.
+ * @return 0 on success, -1 on error.
+ */
+static int
+nc_server_get_endpt(const struct lyd_node *node, struct nc_endpt **endpt, struct nc_bind **bind)
+{
+    uint16_t i;
+    const char *endpt_name;
+
+    if (!node) {
+        ERRARG("node");
+        return -1;
+    }
+
+    while (node) {
+        if (!strcmp(LYD_NAME(node), "endpoint")) {
+            break;
+        }
+        node = (const struct lyd_node *)node->parent;
+    }
+
+    if (!node) {
+        return -1;
```

Comment:
No error message but on other `-1` returns there is one, is it expected or not? Many occurrences.